### PR TITLE
[platform] Implement Hardware Revision Platform API Call for Mellanox Chassis and PSU

### PR DESCRIPTION
**Pull request by robocoder99**
#### Why I did it

This pull request allows calls to be made through the platform 2.0 API that retrieve the PSU and Chassis hardware revision on Mellanox platforms. Access to these values will aid customers in determining their hardware revisions for debugging and technical support. These values are intended to be eventually exposed through the CLI. 

#### How I did it

For the PSU hardware revision I used the existing VPD function calls implemented in https://github.com/Azure/sonic-buildimage/pull/7382

For the Chassis hardware revision I parsed the SMBIOS / DMI type 2 information to retrieve the information. 

#### How to verify it

To verify on a switch open a python 3 interpreter with root access and run the following script....

```
import sonic_platform
platform_chassis = sonic_platform.platform.Platform().get_chassis()

chassis_rev = platform_chassis.get_revision()
psu_rev = platform_chassis.get_all_psus()[0].get_revision()

print("Chassis Rev:", chassis_rev)
print("PSU Rev:", psu_rev)
```

The hardware revision of the chassis and the psu should print successfully. (Generally for mellanox this will look like "A[1-3]") 

Verified on all mellanox platforms aside from MSN2010.

#### Description for the changelog
[platform] Implement Hardware Revision Platform API Call for Mellanox Chassis and PSU

#### A picture of a cute animal (not mandatory but encouraged)
goats

### CODE REVIEW

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 1]
```diff
@@ -343,6 +372,31 @@ def _read_generic_file(self, filename, len):
...
+    def _parse_dmi(self, filename):
+        """
+        Read DMI data chassis data and returns a dictionary of values
+
+        Returns:
+            A dictionary containing the dmi table of the switch chassis info
+        """
+        result = {}
+        try:
+            fileobj = open(filename, "rb")
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:384]: maybe can use "with open" ?

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 2]
```diff
@@ -26,6 +26,23 @@
 
 MLNX_NUM_PSU = 2
 
+DMI_FILE = '/sys/firmware/dmi/entries/2-0/raw'
```
[the remark this answers is not in the document]
Comment (robocoder99) [reply platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:29]: We might be able to. Its seems that using sysfs here is a little more standard than calling commands. This filepath is stable though (it exists  on all mellanox platforms I was able to get my hands on for testing. 

Bulldog 2100
Spider 2410
Panther 2700
Lionfish 3420
Anaconda 3700
Tigris 3800
Ocelot 4410
Tigon 4600
Leopard 4700

[the remark this answers is not in the document]
Comment (robocoder99) [reply platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:29]: The 2-0 indicates the DMI type code I am fairly certain which would be stable, I see no reason the rest of the path would change.

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py (hunk 1 above)
Comment (robocoder99) [reply platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:384]: Fine with me. Was attempting to use the same pattern as in the `read_file()` method above for general consistency.

**Review by keboliu — APPROVED** [review]

**Pull request closed without merging**